File: languages/BTL/Java/Sample.java
// BEGIN ...
package org.softlang.btl;
import org.softlang.btl.Syntax.*;
// END ...
public class Sample {
    public Expr getSample() {
       return
            new Pred(
                new If(
                    new IsZero(new Zero()),
                    new Succ(new Succ(new Zero())),
                    new Zero()));
    }
}

File: languages/BLL/Haskell/Language/BLL/Sample.hs
-- BEGIN ...
module Language.BLL.Sample where
import Language.BLL.Syntax
-- END ...
add = Fix (Lambda "f" (FunType NatType (FunType NatType NatType))
          (Lambda "n" NatType
          (Lambda "m" NatType
            (If (IsZero (Var "n"))
               (Var "m")
               (Succ (Apply (Apply (Var "f") (Pred (Var "n"))) (Var "m"))))))) 
